mobile/library/common/jni/types/JavaVirtualMachine.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include "library/common/jni/import/jni_import.h"
+
+namespace Envoy {
+namespace JNI {
+
+/**
+ * @brief A convenience wrapper for JNI JavaVM type.
+ *
+ */
+class JavaVirtualMachine {
+public:
+  /**
+   * @brief Initializes virtual machine with JavaVM type. It should be called only once.
+   */
+  static void initialize(JavaVM* jvm) { jvm_ = jvm; }
+  static JavaVM* getJavaVM() { return jvm_; }
+  static jint getJNIVersion();
+  static void detachCurrentThread() { jvm_->DetachCurrentThread(); }
```

Comment:
Maybe ASSERT on the return value? 

In general it feels like it would be more C++ to have this be backed by a thread local object that was responsible for calling detach on deinit, that way nothing is responsible for manually calling this

Reply:
Added assert.